src/Microsoft.Fast.Components.FluentUI/Components/NavMenu/FluentNavMenu.razor.cs

```diff
@@ -81,12 +78,12 @@ public partial class FluentNavMenu : FluentComponentBase
 
     public FluentNavMenu()
     {
-        Id = Identifier.NewId();     
+        Id = Identifier.NewId();
     }
 
     internal bool HasSubMenu => _groups.Any();
 
-    internal bool HasIcons => _links.Any(i => i.HasIcon);
+    internal bool HasIcons => _links.Any(i => i.HasIcon) || _groups.Any(g => g.HasNavMenuGutterIcon);
 
```

Comment:
Also needed to take into account NavMenuGroups that have a gutter icon for collapsed NavMenu